### DISCUSSION TITLE
A movie catalog with 60.000 items using SSG, is this a good idea?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [vercel/next.js#18492] [Help] [answered]
Hello guys, i'm working on a Video on Demand project, and i'm thinking about use full SSG, we have 60.000 movies in our catalog, is this a good idea?

I can set a timeout to revalidate each page like I can do when i'm using fallbacks?

Thanks so much!

**Comment by MathiasKandelborg** [comment, marked as answer]
It is possible. The real answer depends on more context, but given a few (1-5) kb data for each page, it shouldn't be problematic.

There are several threads around here with a bunch of discussion surrounding this topic. But the general gist has been long build times (if you choose to build every page before deploy), so a tip I've seen several times is: build the pages on-demand, compared to on build. Here's one of the most relevant threads, the RFC for ISR: https://github.com/vercel/next.js/discussions/11552

Next.js performance in this regard is amazing. I think you should try it out, just don't go bankrupt running it on some build system for 2 weeks to generate 60k pages :+1:

**Reply by wilsonneto-swe** [reply]
Good, thanks so much!
